runescape-client-injector-plugin/src/main/java/net/runelite/injector/MixinInjector.java

```diff
@@ -369,7 +373,86 @@ private void injectMethods(ClassFile mixinCf, ClassFile cf, Map<net.runelite.asm
 		// Handle the rest of the mixin types
 		for (Method method : mixinCf.getMethods())
 		{
-			if (method.getAnnotations().find(INJECT) != null)
+			boolean isClinit = "<clinit>".equals(method.getName());
+			boolean isInit = "<init>".equals(method.getName());
+			boolean hasInject = method.getAnnotations().find(INJECT) != null;
+
+			// You can't annotate clinit, so its always injected
+			if ((hasInject && isInit) || isClinit)
+			{
+				if (!"()V".equals(method.getDescriptor().toString()))
+				{
+					throw new InjectionException("Injected constructors cannot have arguments");
+				}
+
+				Method[] originalMethods = cf.getMethods().stream()
+					.filter(n -> n.getName().equals(method.getName()))
+					.toArray(Method[]::new);
+				// If there isn't a <clinit> already just inject ours, otherwise rename it
+				// This is always true for <init>
+				String name = method.getName();
+				if (originalMethods.length > 0)
+				{
+					name = "rl$$" + (isInit ? "init" : "clinit");
+				}
+
+				Method copy = new Method(cf, name, method.getDescriptor());
+				moveCode(copy, method.getCode());
+				copy.setAccessFlags(method.getAccessFlags());
+				copy.setPrivate();
+				assert method.getExceptions().getExceptions().isEmpty();
+
+				// Remove the call to the superclass's ctor
+				if (isInit)
+				{
+					Instructions instructions = copy.getCode().getInstructions();
+					Instruction[] firstTwo = instructions.getInstructions().stream()
```

Comment:
You can just find the first invokespecial which is guaranteed to be what you want.

Reply:
Is it possible for the first two instructions not to be `aload_0, invokespecal`?

Reply:
Yes if the signature of the super constructor is not ()V. But the first invokespecial is always the super constructor.